core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractCachedThreadReactorSinkProvider.java

```diff
@@ -25,18 +25,39 @@ public abstract class AbstractCachedThreadReactorSinkProvider implements Reactor
 
   private static final int THREAD_CACHE_TIME_LIMIT_IN_MINUTES = 60;
   private static final int TRANSACTION_CACHE_TIME_LIMIT_IN_MINUTES = 10;
+  private boolean sinkIndexEnabled;
 
-  private final Cache<Thread, FluxSink<CoreEvent>> sinks =
+  private final Cache<String, FluxSinkWrapper> sinks =
```

Comment:
If possible, we can add a way to make the keys weak reference so that not to add pressure on memory.